assemblies/static/src/main/resources/static/custom/js/sparkl.js

```diff
@@ -1,423 +1,454 @@
-        var acc = _settings;
-        _.every(arguments, function(el) {
-            acc = acc[el] || undefined;
-            return !!acc;
-        });
-        return acc;
-    };
+    return acc;
+  };
 
+  myself.changeLocation = function (newLocation, bookmarks, isNew) {
+    if (!newLocation) {
+      return;
+    }
 
-    myself.changeLocation = function(newLocation, bookmarks, isNew) {
-        if (!newLocation) {
-            return;
-        }
-        var hash = (bookmarks && !_.isEmpty(bookmarks)) ? '#' + generateHashValue("bookmark", {
-            impl: "client",
-            params: bookmarks
-        }) : "";
-        if (isNew) {
-            window.open(newLocation + hash);
-        } else {
-            window.location = newLocation + hash;
-        }
-    };
+    var hash = (bookmarks && !_.isEmpty(bookmarks)) ? '#' + generateHashValue("bookmark", {
+      impl: "client",
+      params: bookmarks
+    }) : "";
 
-    function generateHashValue(key, value) {
-        var obj = Dashboards.getHashValue(),
-            json;
-        if (arguments.length == 1) {
-            obj = key;
-        } else {
-            obj[key] = value;
-        }
-        json = JSON.stringify(obj);
-        return json;
+    if (isNew) {
+      window.open(newLocation + hash);
+    } else {
+      window.location = newLocation + hash;
     }
+  };
 
-    myself.isValidName = function(name, type) {
-        var reg = _settings.expressions[type || 'plugin'] || /.*/;
-        return reg.test(name);
-    };
+  function generateHashValue(key, value) {
+    var obj = Dashboards.getHashValue();
 
+    if (arguments.length === 1) {
+      obj = key;
+    } else {
+      obj[key] = value;
+    }
 
-    myself.isUpdated = function(srcVersion, pluginVersion) {
-        return (srcVersion == pluginVersion);
-    };
+    return JSON.stringify(obj);
+  }
 
+  myself.isValidName = function (name, type) {
+    var reg = _settings.expressions[type || 'plugin'] || /.*/;
 
-    myself.isJobError = function(json) {
-        return (json && json.result === false);
-    };
+    return reg.test(name);
+  };
+
+  myself.isUpdated = function (srcVersion, pluginVersion) {
+    var srcVersionObj = myself.parseVersion(srcVersion);
+    var pluginVersionObj = myself.parseVersion(pluginVersion);
+
+    if (pluginVersionObj.major < srcVersionObj.major) return false;
+    if (pluginVersionObj.minor < srcVersionObj.minor) return false;
+    if (pluginVersionObj.other < srcVersionObj.other) return false;
+    if (pluginVersionObj.patch < srcVersionObj.patch) return false;
 
-    myself.hasUploaderComponent = function() {
-        return !(typeof FileUploaderComponent == 'undefined');
+    return true;
+  };
+
+  myself.parseVersion = function (version) {
+    // Matches - Major.minor.other.patch
+    var versionRegx = /^([0-9]+)(?:\.([0-9]*)(?:\.([0-9]*))?(?:\.([0-9]*))?)?/;
+
+    var match = versionRegx.exec(version);
+
+    return {
+      major: Number(match[1]),
```

Comment:
The regex is very permissive and matches a lot of combinations...
Things like `"1"` match and result in `match[2]` and following entries to be `undefined`. Then, `Number(undefined)` results in `NaN`.
As such, it would be best to do `Number(match[i] || "")`.